gpu/features/test/test_normals.cpp

```diff
@@ -371,6 +371,93 @@ TEST(PCL_FeaturesGPU, normals_highlevel_4)
     }
 }
 
+// Test from issue:
+// - https://github.com/PointCloudLibrary/pcl/issues/2371#issuecomment-577727912
+TEST(PCL_FeaturesGPU, issue_2371)
+{
+    // This number is magic, do not set to lower value.
+    // It may affect error reproducibility.
+    const std::size_t N = 1000;
+    std::vector<pcl::PointXYZ> cloud_cpu(N, {0.0, 0.0, 0.0});
+
+    pcl::gpu::NormalEstimation::PointCloud cloud_gpu;
+    cloud_gpu.upload(cloud_cpu);
+
+    pcl::gpu::NormalEstimation ne_gpu;
+    ne_gpu.setInputCloud(cloud_gpu);
+
+    const float radius_search = 2.0F;
+    const int max_results = 500;
+    ne_gpu.setRadiusSearch(radius_search, max_results);
+
+    pcl::gpu::NormalEstimation::Normals normals_gpu;
+    ne_gpu.compute(normals_gpu);
+}
+
+TEST(PCL_FeaturesGPU, normals_nan_gpu)
+{
+    const std::size_t N = 5;
+
+    PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>());
+    cloud->points.assign(N, {0.0, 0.0, 0.0});
+
+    const float radius_search = 2.0F;
+    const int max_results = 500;
+
+    pcl::gpu::NormalEstimation::PointCloud cloud_device;
+    cloud_device.upload(cloud->points);
+
+    pcl::gpu::NormalEstimation ne_device;
+    ne_device.setInputCloud(cloud_device);
+    ne_device.setRadiusSearch(radius_search, max_results);
+
+    pcl::gpu::NormalEstimation::Normals normals_device;
+    ne_device.compute(normals_device);
+
+    std::vector<PointXYZ> downloaded;
+    normals_device.download(downloaded);
+
+    ASSERT_EQ(downloaded.size(), N);
+
+    for(std::size_t i = 0; i < downloaded.size(); ++i)
+    {
+        const PointXYZ& n = downloaded[i];
+
+        ASSERT_TRUE(std::isnan(n.x));
+        ASSERT_TRUE(std::isnan(n.y));
+        ASSERT_TRUE(std::isnan(n.z));
+    }
+}
+
+TEST(PCL_FeaturesGPU, normals_nan_cpu)
```

Comment:
This test doesn't belong in GPU test series. The CPU test belongs in the main test suite.

For the GPU test suite, there are 2 options (no need to change. just specifying for clarification):
* test for equality between results of GPU and CPU on common operations
* test GPU specific functionality independently here (current version, but the functionality is common)

Reply:
Okay, so we need:
- keep `TEST(PCL_FeaturesGPU, issue_2371)` here
- remove `TEST(PCL_FeaturesGPU, normals_nan_cpu)` from `test_normals.cpp`
- add `TEST(PCL_FeaturesGPU, normals_nan_gpu)` in another pull request

Is my understanding correct? :)

Reply:
Done

Reply:
> add TEST(PCL_FeaturesGPU, normals_nan_gpu) in another pull request

See https://github.com/PointCloudLibrary/pcl/pull/3743